libs/langchain-community/src/vectorstores/astradb.ts

```diff
@@ -40,18 +49,35 @@ export class AstraDBVectorStore extends VectorStore {
 
   private readonly contentKey: string; // if undefined the entirety of the content aside from the id and embedding will be stored as content
 
+  private readonly batchSize: number;
+
+  caller: AsyncCaller;
+
   _vectorstoreType(): string {
     return "astradb";
   }
 
   constructor(embeddings: EmbeddingsInterface, args: AstraLibArgs) {
     super(embeddings, args);
 
-    this.astraDBClient = new AstraDB(args.token, args.endpoint);
-    this.collectionName = args.collection;
-    this.collectionOptions = args.collectionOptions;
-    this.idKey = args.idKey ?? "_id";
-    this.contentKey = args.contentKey ?? "text";
+    const {
+      token,
+      endpoint,
+      collection,
+      collectionOptions,
+      idKey,
+      contentKey,
+      batchSize,
+      ...callerArgs
+    } = args;
```

Comment:
This is missing `namespace` from `AstraLibArgs`, don't want that getting included in `callerArgs`.

Side note, it doesn't look like this was implemented anywhere previously. Is it needed?

Reply:
I removed namespace for now since it is unused but will be adding it in the future when the new version of TS client is ready.

Batching wasn't implemented before but it is needed as the `insertMany` has a cap of 20 documents